### ISSUE TITLE
Is it enough to use out-of-sample predictions to avoid overfitting?

### ISSUE THREAD

**Issue by rgalljamov** [cleanlab/cleanlab#1160]
Hey Cleanlab Team,

Thank you for your incredible work! It's great to see such focused efforts on cleaning data!

I’m working on a binary image classification problem and have a dataset with an unknown but high portion of mislabeled data (might be up to 40%). It’s expensive to obtain clear labels. So, I followed an approach similar to what I think you suggest and iteratively cleaned the dataset. After five iterations, I improved accuracy from 64% to 95%. However, when I tested the model on a separate test set, I saw significant overfitting.

After reading your documentation, I realized that it's crucial to use out-of-sample predictions. This makes total sense, and I've cleaned the entire dataset which included the training data. However, I am wondering: 
**Is it enough to use out-of-sample predictions to avoid overfitting?**

Here’s why I ask: Unlike in standard overfitting, where we overfit the model to the data, **we overfitted the data to the model**, right? Instead of adjusting model's parameters to a fixed dataset, we've adjusted the dataset to a fixed model. And this should still happen even when we use out-of-sample predictions, right? By cleaning the dataset, we adjust it to the model. Then we train the same model on the cleaned dataset and adjust the dataset a second time to the model. Wouldn't we reinforce potential biases of the model this way? 

On the other hand, by using cross-validation, we are kind of training an ensemble of models, which helps against overfitting. And in addition, we might also use a less powerful model, like resnet18 instead of convnext-large?

Thank you very much in advance for any idea and advice!
Best regards,
Rustam

**Comment by jwmueller** [comment]
There are many subtle complexities in this topic, so we just made a dedicated tutorial about it:
https://docs.cleanlab.ai/master/tutorials/improving_ml_performance.html

That should hopefully answer your questions, but feel free to ask follow-ups!
